### DISCUSSION TITLE
Add Folder

### DISCUSSION THREAD

**Discussion by randy-roberts** [chhoumann/quickadd#687] [Ideas]
I would like to see a feature to add a folder via quickadd. I want to use it in a macro like the new note functionality, where I can provide a set of folders, and the action creates a sub-folder in the one selected. 
Are there any plans to add this functionality?

**Comment by chhoumann** [comment]
Coming back to this one late - sorry about that. Good news: what you describe works natively now.

Since you posted this, folder paths gained full format syntax support, and the folder picker lets you type a new path to create it on the spot. For your macro use case:

- Set the Template choice to **In a specific folder** and list your candidate folders - running the choice shows a folder suggester, and you can type a new sub-path in the picker to create it inline.
- Or make the sub-folder part of the path itself: a folder path like `Projects/{{VALUE:client}}/{{DATE:YYYY}}` prompts for the value and auto-creates any missing folders when the note is created.

If you ever need an empty folder with no note at all, a tiny user script step in the macro does it:

```js
module.exports = async ({ app, quickAddApi }) => {
  const parent = await quickAddApi.suggester((f) => f, ["Area1", "Area2"]);
  const name = await quickAddApi.inputPrompt("Sub-folder name");
  await app.vault.createFolder(`${parent}/${name}`);
};
```

The folder settings and path format syntax are documented here: https://quickadd.obsidian.guide/docs/Choices/TemplateChoice
